helpers/sql-obfuscation/lib/opentelemetry/helpers/sql_obfuscation.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0module OpenTelemetry
+
+require 'opentelemetry-common'
+
+module OpenTelemetry
+  module Helpers
+    #
+    # This class contains SQL obfuscation behavior to share with
+    # instrumentation for specific database adapters.
+    # The class uses code from: https://github.com/newrelic/newrelic-ruby-agent/blob/1fca78cc7a087421ad58088d8bea72c0362bc62f/lib/new_relic/agent/database/obfuscation_helpers.rb
+    #
+    # To use this in your instrumentation, the `Instrumentation` class for
+    # your gem must contain configuration options for:
+    #  * `:db_statement`
+    #    Example:
+    #    `option :db_statement, default: :include, validate: %I[omit include obfuscate]`
+    #  * `:obfuscation_limit`
+    #    Example:
+    #    `option :obfuscation_limit, default: 2000, validate: :integer`
+    #
+    # If you want to add support for a new adapter, update the following
+    # constants to include keys for your adapter:
+    #  * DIALECT_COMPONENTS
+    #  * CLEANUP_REGEX
+    # You must also add a new constant that uses `generate_regex` with your
+    # adapter's dialect components that is named like
+    # `<ADAPTER>_COMPONENTS_REGEX`, such as: `MYSQL_COMPONENTS_REGEX`.
+    #
+    # @api public
+    class SqlObfuscation
```

Comment:
if these are never instantiated perhaps this can be a module with the methods listed under `module_function`

Reply:
Good idea! I'm still wrapping my head around when to choose `module_function` over `extend self`. 

It looks like the repo has more uses of `extend self`(17) than `module_function`(3). Also, [Rubocop defaults to `module_function` over `extend self`.](https://docs.rubocop.org/rubocop/cops_style.html#stylemodulefunction) I know they have slightly different behavior too. 

What's your reasoning for using `module_function` instead of `extend self`?

Reply:
> This is the main difference between extend self and module_function. The latter one really creates a module function by restricting access to the included method while extend self still allows access to the included method.

https://medium.com/rubycademy/is-module-function-really-the-same-as-extend-self-ac1e96a1cda0

Reply:
I don't have a strong preference but just pointing out that if our intent is to treat these like global helpers and not treated as mixins, then module_function makes that more explicit. 

If we want more flexibility in composition, then we should keep using extend self

Reply:
Thanks, Ariel! That's very helpful!

Reply:
Updated in: 4e8aea4ae22972757ff1952bf3eea0ad9ba5b301